src/features/device-types/storage/s3.ts

```diff
@@ -66,11 +81,16 @@ async function getFileInfo(s3Path: string) {
 }
 
 export async function getFile(s3Path: string) {
-	const req = s3Client.getObject({
-		Bucket: S3_BUCKET,
-		Key: s3Path,
-	});
-	return await req.promise();
+	try {
+		const req = s3Client.getObject({
+			Bucket: S3_BUCKET,
+			Key: s3Path,
+		});
+		return await req.promise();
+	} catch (err) {
+		// catch errors for private device types when running unauthenticated
+		logUnauthenticatedWarning(s3Client, s3Path, err);
```

Comment:
@ab77 This atm silences ALL errors. Shouldn't we re-throw the error when this is a different type of error that logUnauthenticatedWarning didn't handle?

Reply:
Fair point, we should make sure other errors aren't ignored